<launch>

  <!-- Define the use_simulator argument-->
  <arg name="use_simulator" default="false" />

  <!-- Define the calibration_type argument, default to static -->
  <arg name="calibration_type" default="static" />

  <!-- Start FT process node -->
  <node name="ft_process" pkg="virtual_tactile_pad" type="ft_process_node.py" output="screen">
    <param name="use_simulator" value="$(arg use_simulator)" />
    <param name="calibration_type" value="$(arg calibration_type)" />
  </node>

  <!-- Start digital recognition node -->
  <node name="data_saver" pkg="virtual_tactile_pad" type="data_saver_node.py" output="screen">
    <param name="use_simulator" value="$(arg use_simulator)" />
  </node>

  <!-- Load robot description -->
  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find virtual_tactile_pad)/urdf/panda_pad.urdf" />

  <!-- Start robot state publisher -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

  <!-- Start joint state publisher with GUI (for testing) -->
  <node name="joint_state_publisher_gui" pkg="joint_state_publisher_gui" type="joint_state_publisher_gui" />

  <!-- Start contact visualizer node -->
  <node name="ft_visualize" pkg="virtual_tactile_pad" type="visualize_node.py" output="screen" />

  <!-- Start RViz -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find virtual_tactile_pad)/rviz/ft_config.rviz" />

</launch>
